ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1152,19 +1155,53 @@ private void updatePartitionResourcesMappingFromExternalView(Collection<External
         logger.warn(
             "Partition to resource mapping for individual dc (non-aggregated view) would be built from ideal states");
       } else {
+        Map<String, Map<String, String>> partitionToDCToResourceNames = new HashMap<>();
```

Comment:
Just curious why we switch the ConcurrentHashMap to HashMap?

Reply:
I mistakenly used ConcurrentHashMap earlier here. I think HashMap is probably sufficient here since we are serially adding all entries to it.

Reply:
I will keep it this way and update it later.